tests/test_yaml/test2.py

```diff
@@ -16,4 +16,4 @@ def prepare(a, b):
     print(f"this is fake method {a} {b}")
 
 
-valohai.prepare(step="foobar1", default_parameters=params, default_inputs=inputs)
+valohai.prepare(step="foobar1", default_parameters=params, default_inputs=inputs, image="hope/thiswillnotchange:1234")
```

Comment:
I would like to see this test pass without adding the image to test2.py.

If user has an original `valohai.yaml` with `image: hope/thiswillnotchange:1234` and then doesn't have `image` set in their `prepare()` call, then we should let the original `image: hope/thiswillnotchange:1234` be untouched.

As a rule of thumb: Anything that is **not** defined in `.py` file should stay as it is when merging.

This test2 should test for that behavior.

Reply:
Changed 🥷 ✅